sklearn/base.py

```diff
@@ -578,3 +580,39 @@ def is_regressor(estimator):
         True if estimator is a regressor and False otherwise.
     """
     return getattr(estimator, "_estimator_type", None) == "regressor"
+
+
+def frozen_fit(estimator, method, X, y, **kwargs):
+    """Fit the estimator if not frozen, and return the result of method
+
+    A frozen estimator has an attribute ``frozen`` set to True
```

Comment:
How about naming this "fit_if_not_frozen". Maybe that would help @amueller (and others) understand.